docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,184 @@
+    * allows monitoring the health of devices
+    * helps troubleshooting when problems with the device are reported
+
+[^1]: more details see appendix [Device Domain](#device-domain)
+
+## Child-Devices
+
+**thin-edge** facilitates IoT functionality to the device it is running on, as well as to devices that are connected to that device.
+  * the device **thin-edge** is running on is referred as the **main-device**
+    * **thin-edge** on the **main-device** establishes and manages all communication to the cloud
+  * all devices connected to the **main-device** are referred as **external child-devices** 
+  * each **external child-device** can be represented in the cloud with its individual **device twin**
+    * a unique **child-id** makes the association between each **external child-device** and it's **device twin**
+  * all **telemetry data** and **device management** functionality can appear in the context of the 
+    **external child-device's** individual **device twin**, or the **main-device's** **device twin**
+  * also _containers_ or _processes_ running on the **main-devices** can be handled like **external child-device**;
+    those are referered then as **logical child-devices**
+
+The figure below illustrates the child-device concept.
+
+![Device Concept](images/device-concept.svg)
+
+
+## Plugins
+
+**thin-edge** realizes **Device Management** based on **plugins**.
+  * a **plugin** can be an (external) executable (e.g. as the `c8y_configuration_plugin` for _configuration management_)
+    or a thin-edge built-in software component (e.g. as for _software management_)
+  * to realize **Device Management** functionality, a **plugin** accesses _resources_ and _services_ of the device, as e.g.
+      * _software management_ uses the device's _package manager_
+      * _configuration management_ reads/writes device's _configuration files_
+  * a **plugin** can facilitate **Device Management** functionality for the **main-device** and **external child-device**, but runs usually only on the **main-device**
+    * _resources_ and _services_ (e.g. file system, package manager, ...) of the **main-device** can be directly accessed by the **plugin**, as it runs on the **main-device**
+    * all _resources_ and _services_ of an **external child-device** are only remotely accessible for the **main-device**
+      * the **plugin** running on the **main-device** needs another software component that establish the remote access for those _resources_ and _services_
+      * that software component is referred as **child-device agent**
+  * a **plugin** can also facilitate **Device Management** functionality for containers running on the **main-device**
+    * similar to **external child-devices**, all _resources_ and _services_ of a container are accessed by a **plugin** via a **child-device agent** that runs inside the corresponding container 
```

Comment:
Having in mind that a container is a **logical child-device**, it might not sound so weird. But I agree, for one who has strong focus on containers and does not deal with child-devices at all, it might be irritating.

However I was already from beginning not really fine with the term **... "agent"**. And now with **plugin agent** it gives even less a clue what that thing is about.

What do you think about the term **plugin adapter**? In software an adapter "provides a compatible connection between the components of a system". And that is exactly what our piece of software is doing.

@albinsuresh , what do you think about that?

Reply:
I renamed now "child device agent" to "plugin adapter".

See https://github.com/thin-edge/thin-edge.io/pull/1554/commits/c75a26ddc532bd3ef547bcc1071feaf7ef047e28